File: app/src/main/java/br/edu/ifspsaocarlos/agenda/activity/MainActivity.java
package br.edu.ifspsaocarlos.agenda.activity;

import android.content.Intent;
import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.view.View;

import br.edu.ifspsaocarlos.agenda.R;

public class MainActivity extends BaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        FloatingActionButton myFab = (FloatingActionButton) findViewById(R.id.fab);
        myFab.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                Intent i = new Intent(getApplicationContext(), DetalheActivity.class);
                startActivityForResult(i, 0);

            }
        });

        buildListView();
    }

    @Override
    protected void onRestart() {
        super.onRestart();
        buildListView();
        searchView.clearFocus();

    }
}

File: app/src/main/java/br/edu/ifspsaocarlos/agenda/activity/MainActivity.kt
package br.edu.ifspsaocarlos.agenda.activity

import android.content.Intent
import android.os.Bundle
import br.edu.ifspsaocarlos.agenda.R
import com.google.android.material.floatingactionbutton.FloatingActionButton

class MainActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val mainFab = findViewById<FloatingActionButton>(R.id.fab)
        mainFab.setOnClickListener {
            val i = Intent(applicationContext, DetailActivity::class.java)
            startActivity(i)
        }

        val secondaryFab = findViewById<FloatingActionButton>(R.id.fabProvider)
        secondaryFab.setOnClickListener {
            val i = Intent(applicationContext, ContentProviderActivity::class.java)
            startActivity(i)
        }

        buildListView()
    }

    override fun onRestart() {
        super.onRestart()
        buildListView()
        invalidateOptionsMenu()
    }
}